tony-history-server/app/utils/ParserUtils.java

```diff
@@ -30,28 +32,52 @@
 public class ParserUtils {
   private static final Logger.ALogger LOG = Logger.of(ParserUtils.class);
 
-  public static JobMetadata parseMetadata(FileSystem fs, Path metadataFilePath) {
-    if (!pathExists(fs, metadataFilePath)) {
-      return new JobMetadata();
+  @VisibleForTesting
+  static boolean isValidMetadata(String[] metadata, String jobIdRegex) {
```

Comment:
This method is hard to understand as is. I suggest renaming it to `isValidJobHistoryFileName` and pass in the full job history file name directly.

Some Javadoc with an example would also be helpful.

Reply:
I agree with your suggestion for the function name; however, I think we still have to do the same thing once we have the history file name: splitting and storing them in an array and check each element of the array to see if it is valid. I think it's hard to understand because of the way I format the code. How about something like this: https://gist.github.com/pdtran3k6/3b427fc1337c58a494c9b161c58c98c6

Reply:
What I meant is: I don't think this method should take in a String array. It should take in the full file name and do the parsing/validation internally. Don't you think it's a much cleaner API to just pass a full file name and have a method do validation on it, than for an API to expect the caller to pre-split the filename first?

Please also add some Javadoc.

Reply:
Agreed. Fixed: https://github.com/linkedin/TonY/pull/81/commits/ed06cf126ca618f794531e647344310f689b6969.